frontend/src/v5/store/tickets/tickets.sagas.ts

```diff
@@ -60,14 +65,152 @@ export function* fetchTicket({ teamspace, projectId, modelId, ticketId, isFedera
 	} catch (error) {
 		yield put(DialogsActions.open('alert', {
 			currentActions: formatMessage(
-				{ id: 'tickets.fetchTicket.error', defaultMessage: 'trying to fetch the ticket details for {model} ' },
+				{ id: 'tickets.fetchTicket.error', defaultMessage: 'trying to fetch the ticket details for {model}' },
 				{ model: isFederation ? 'federation' : 'container' },
 			),
 			error,
 		}));
 	}
 }
 
+export function* fetchTicketComments({
+	teamspace,
+	projectId,
+	modelId,
+	ticketId,
+	isFederation,
+}: FetchTicketCommentsAction) {
+	try {
+		const fetchModelTicketComments = isFederation
+			? API.Tickets.fetchFederationTicketComments
+			: API.Tickets.fetchContainerTicketComments;
+		const { comments } = yield fetchModelTicketComments(teamspace, projectId, modelId, ticketId);
+		const richComments = sortBy(comments, 'createdAt').map(({ createdAt, updatedAt, ...comment }) => ({
+			...comment,
+			createdAt: new Date(createdAt),
+			updatedAt: new Date(updatedAt),
+		}));
+
+		yield put(TicketsActions.fetchTicketCommentsSuccess(modelId, ticketId, richComments));
+	} catch (error) {
+		yield put(DialogsActions.open('alert', {
+			currentActions: formatMessage(
+				{ id: 'tickets.fetchTicketComments.error', defaultMessage: 'trying to fetch the comments for {model} ticket' },
+				{ model: isFederation ? 'federation' : 'container' },
+			),
+			error,
+			details: formatMessage({
+				id: 'tickets.fetchTicketComments.error.details',
+				defaultMessage: 'If reloading the page doesn\'t work please contact support',
+			}),
+		}));
+	}
+}
+
+export function* createTicketComment({
+	teamspace,
+	projectId,
+	modelId,
+	ticketId,
+	isFederation,
+	comment,
+	onSuccess,
+	onError,
+}: CreateTicketCommentAction) {
+	try {
+		const createModelTicketComment = isFederation
+			? API.Tickets.createFederationTicketComment
+			: API.Tickets.createContainerTicketComment;
+		const data = yield createModelTicketComment(teamspace, projectId, modelId, ticketId, comment);
```

Comment:
The `createModelTicketComment` is the API call to actually create a comment in the backend.
The  `upsertTicketCommentSuccess` instead, is used to save the comment in redux.

When `createModelTicketComment`, the only thing that you get back is the `_id` property of the ticket, which is then used to update the ticket created in the frontend before storing it into redux. As for the two dates, they are created (and added) only after the POST request is successful, so that they are as close to the values in the backend as possible.